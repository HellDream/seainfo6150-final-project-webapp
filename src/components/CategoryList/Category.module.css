.categoryContainer {
    display: grid;
    grid-template-columns: 1fr;
    column-gap: 2.5rem;
    row-gap: 4rem;
    padding: 5rem 5rem 3rem 5rem;
    list-style-type: none;
}


@media screen and (min-width: 700px) {
    .categoryContainer {
      grid-template-columns: repeat(2, 1fr);
    }
  }

@media screen and (min-width: 1150px) {
    .categoryContainer {
        grid-template-columns: repeat(3, 1fr);
    }
}