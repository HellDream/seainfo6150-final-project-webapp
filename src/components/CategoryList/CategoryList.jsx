import React from 'react';
import CategoryItem from '../categoryItem/CategoryItem';
import styles from './Category.module.css';
import electronicImg from '../../images/electronics.jpeg';
import booksImg  from '../../images/books.jpg';
import funituresImg from '../../images/furnitures.jpg';
import entertainImg from '../../images/entertainments.png';
import toolsImg from '../../images/tools.jpg';
import others from '../../images/others.jpg';
const categories = [
    {
        slug: 0,
        title: 'books',
        imageURI: booksImg,
    },
    {
        slug: 1,
        title: 'electronics',
        imageURI: electronicImg,
    },
    {
        slug: 2,
        title: 'furnitures',
        imageURI: funituresImg,
    },

    {
        slug: 3,
        title: 'entertainments',
        imageURI: entertainImg,
    },
    {
        slug: 4,
        title: 'tools',
        imageURI: toolsImg,
    },

    {
        slug: 5,
        title: 'others',
        imageURI: others,
    },
];
const CategoryList = () => {
    return (
        <ul className={styles.categoryContainer}>
            {categories.map((category) => (
                <CategoryItem category={category} key={category.slug} />
            ))}
        </ul>
    );
};
export default CategoryList;
