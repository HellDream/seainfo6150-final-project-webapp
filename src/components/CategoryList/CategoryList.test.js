import React from 'react';
import CategoryList from './CategoryList';
import { render } from '@testing-library/react'

describe('CategoryList tests', () => {
    it('renders correctly', () => {
        const { container } = render(<CategoryList />);
        expect(container).toMatchSnapshot();
    });
});
