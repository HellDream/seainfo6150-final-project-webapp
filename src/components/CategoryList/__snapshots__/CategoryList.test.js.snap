// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList tests renders correctly 1`] = `
<div>
  <ul
    class="categoryContainer"
  >
    <li
      class="category"
    >
      <a
        class="link"
        href="/books"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="undefined"
            class="image"
            src="books.jpg"
          />
        </div>
        <h3
          class="title"
        >
          books
        </h3>
      </a>
    </li>
    <li
      class="category"
    >
      <a
        class="link"
        href="/electronics"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="undefined"
            class="image"
            src="electronics.jpeg"
          />
        </div>
        <h3
          class="title"
        >
          electronics
        </h3>
      </a>
    </li>
    <li
      class="category"
    >
      <a
        class="link"
        href="/furnitures"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="undefined"
            class="image"
            src="furnitures.jpg"
          />
        </div>
        <h3
          class="title"
        >
          furnitures
        </h3>
      </a>
    </li>
    <li
      class="category"
    >
      <a
        class="link"
        href="/entertainments"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="undefined"
            class="image"
            src="entertainments.png"
          />
        </div>
        <h3
          class="title"
        >
          entertainments
        </h3>
      </a>
    </li>
    <li
      class="category"
    >
      <a
        class="link"
        href="/tools"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="undefined"
            class="image"
            src="tools.jpg"
          />
        </div>
        <h3
          class="title"
        >
          tools
        </h3>
      </a>
    </li>
    <li
      class="category"
    >
      <a
        class="link"
        href="/others"
      >
        <div
          class="imageBlock"
        >
          <img
            alt="undefined"
            class="image"
            src="others.jpg"
          />
        </div>
        <h3
          class="title"
        >
          others
        </h3>
      </a>
    </li>
  </ul>
</div>
`;
